src/Storages/Kafka/KafkaBlockInputStream.cpp

```diff
@@ -140,6 +140,13 @@ Block KafkaBlockInputStream::readImpl()
 
         if (new_rows)
         {
+            // In read_kafka_message(), ReadBufferFromKafkaConsumer::nextImpl()
+            // will be called, that may make something unusable, i.e. clean
+            // ReadBufferFromKafkaConsumer::messages, which is accessed from
+            // ReadBufferFromKafkaConsumer::currentTopic() (and other helpers).
+            if (buffer->polledDataUnusable())
```

Comment:
actually if it's stalled you can access currentTopic etc.

Reply:
How? Let's say that buffer is in REBALANCE_HAPPENED state, so no messages, and it will try to access them via iterator to get the topic

Reply:
Then if branch above that line will catch it.

Reply:
> actually if it's stalled you can access currentTopic etc.

actually if it's stalled you >>>can't<<< access currentTopic etc.

Silly error. I wanted to say "check isStalled indead of polledDataUnusable"